node/Utils/CheckRLimits.hsc

```diff
@@ -0,0 +1,58 @@
+{-# language ScopedTypeVariables #-}
+{-# language TypeApplications #-}
+{-# language CApiFFI #-}
+
+module Utils.CheckRLimits(checkRLimits) where
+
+import Control.Exception
+import Control.Monad
+import Foreign hiding (void)
+import Foreign.C.Error
+import Foreign.C.Types
+import GHC.Stack
+import System.Exit
+import System.IO
+
+-- RLIMIT_NOFILE
+
+#include <sys/resource.h>
+#include <errno.h>
+#include <sys/types.h>
+
+-- k = #size rlimit
+
+foreign import ccall "sys/resource.h getrlimit" 
+    c_getrlimit :: CInt -> Ptr () -> IO CInt
+
+foreign import ccall "sys/resource.h setrlimit" 
+    c_setrlimit :: CInt -> Ptr () -> IO CInt
+
+getOpenFileLimits :: HasCallStack => IO ((#type rlim_t), (#type rlim_t))
+getOpenFileLimits = allocaBytes (#size struct rlimit) $ \ptr -> do
+    err <- c_getrlimit (#const RLIMIT_NOFILE) ptr
+    if err /= 0 then do
+        Errno errno <- getErrno
+        error $ "getOpenFileLimits: errno not equal to 0: " <> show errno
```

Comment:
This error condition makes me worried that we'll encounter an environment that, for some reason, errors out on the `getrlimit` call, but actually still has a high enough file handle limit. In that case this mandatory check will be a big inconvenience to the user who could otherwise run the node without any issues.